storage/innobase/dict/dict0dd.cc

```diff
@@ -1080,7 +1080,6 @@ dberr_t dd_rename_tablespace(dd::Object_id dd_space_id,
   }
 
   bool fail = client->update(new_space);
-  ut_ad(!fail);
```

Comment:
Why this assert is removed?

Reply:
Because in test : encryption.innodb-bad-key-change2 I am testing : RENAME TABLE t1 TO t1new. t1 is encrypted and keyring with incorrect encryption key is loaded. Decryption of t1 fails and client->update fails. The path where client->update can fail were not tested prior to this test.

Reply:
Can you retain the assert and add checks if tablespace is readable (decryptable) or not early before doing DD update?

Relying on DD code to know decryption failure isn't recommended. Generally DD calls shouldn't fail and the assert should be retained.